std/string.d

```diff
@@ -6046,6 +6084,48 @@ bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S)
     assert(!isNumeric("+"));
 }
 
+/**
+ * Helper function for isNumeric, ASCII only case insensitive
+ * string comparison.
+ * Returns:
+ *     `true` if both strings are the same
+ */
+private bool asciiCmp(S1, S2)(S1 str1, S2 str2)
+    if (isInputRange!S1 &&
+        isInputRange!S2 &&
+        isSomeChar!(ElementEncodingType!S1) &&
+        isSomeChar!(ElementEncodingType!S2) &&
+        !isInfinite!S2 &&
+        !isInfinite!S2)
+{
+    import std.ascii : toLower;
+
+    for (;;)
+    {
+        if (str1.empty)
+            return str2.empty ? true : false;
+        if (str2.empty)
+            return false;
+
+        if (toLower(str1.front) != toLower(str2.front))
+            return false;
+
+        str1.popFront;
+        str2.popFront;
+    }
+}
```

Comment:
This all is equivalent to `str1.map!toLower.equal(str2.map!toLower)`, right? And since this is just a helper for `byCodeUnit`, I suggest putting it there as a local function with fixed types, like so:

```d
    // ASCII only case insensitive comparison with a string
    static bool asciiCmp(typeof(codeUnits) a, string b)
    {
        import std.algorithm : equal, map;
        import std.ascii : toLower;
        return a.map!toLower.equal(b.byChar.map!toLower);
    }
```